drivers/serial/usart_gd32.c

```diff
@@ -124,27 +146,189 @@ static int usart_gd32_err_check(const struct device *dev)
 	return errors;
 }
 
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+int usart_gd32_fifo_fill(const struct device *dev, const uint8_t *tx_data,
+			 int len)
+{
+	const struct gd32_usart_config *const cfg = dev->config;
+	uint8_t num_tx = 0U;
+
+	while ((len - num_tx > 0) &&
+		usart_flag_get(cfg->reg, USART_FLAG_TBE)) {
+		usart_data_transmit(cfg->reg, tx_data[num_tx++]);
+	}
+
+	return num_tx;
+}
+
+int usart_gd32_fifo_read(const struct device *dev, uint8_t *rx_data,
+			 const int size)
+{
+	const struct gd32_usart_config *const cfg = dev->config;
+	uint8_t num_rx = 0U;
+
+	while ((size - num_rx > 0) &&
+		usart_flag_get(cfg->reg, USART_FLAG_RBNE)) {
+		rx_data[num_rx++] = usart_data_receive(cfg->reg);
```

Comment:
while expression uses same indentation as loop code, difficulties readability. `usart_flag_get` should align to second `(` (7 spaces)

Reply:
aligned to second '('.